core/src/main/scala/com/avsystem/commons/concurrent/TaskExtensions.scala

```diff
@@ -0,0 +1,57 @@
+package com.avsystem.commons
+package concurrent
+
+import com.avsystem.commons.concurrent.TaskExtensions.{TaskCompanionOps, TaskOps}
+import com.avsystem.commons.misc.Timestamp
+import monix.eval.Task
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration.FiniteDuration
+
+trait TaskExtensions {
+  implicit def taskOps[T](task: Task[T]): TaskOps[T] = new TaskOps(task)
+
+  implicit def taskCompanionOps(task: Task.type): TaskCompanionOps.type = TaskCompanionOps
+}
+
+object TaskExtensions extends TaskExtensions {
+  final class TaskOps[T](private val task: Task[T]) extends AnyVal {
+    /**
+      * Like regular `timeout` but [[TimeoutException]] is created lazily (for performance).
+      */
+    def lazyTimeout(after: FiniteDuration, msg: => String): Task[T] =
+      task.timeoutTo(after, Task.raiseError(new TimeoutException(msg)))
+
+    /**
+      * Similar to [[Task.tapEval]], accepts simple consumer function as an argument
+      */
+    def tapL(f: T => Unit): Task[T] =
+      task.map(_.setup(f))
+
+    /**
+      * Similar to [[Task.tapError]], accepts [[PartialFunction]] as an argument
+      */
+    def tapErrorL[B](f: PartialFunction[Throwable, B]): Task[T] =
+      task.tapError(t => Task(f.applyOpt(t)))
+  }
+
+  object TaskCompanionOps {
```

Comment:
Can we please do `traverseMap`, or similar, here? I've had my fair share of `Task.traverse(map.toSeq)(...).map(_.toMap)`

Reply:
What signature to you have in mind? `def traverseMap[K, A, B](map: Map[K, A])(f: A => Task[B]): Task[Map[K, B]]` ?

Reply:
I was thinking `def traverseMap[X, Y, K, V](map: Map[X, Y])(f: (X, Y) => Task[(A, B)]): Task[Map[A, B]]`, but yours seems useful too - though maybe the method should also get the key value (`f: (K, A) => Task[B]`)?

Reply:
Why not both? :smile: